src/test/scala/strawman/collection/test/WithFilterTest.scala

```diff
@@ -0,0 +1,58 @@
+package strawman
+package collection
+package test
+
+import immutable.{List, TreeSet, TreeMap}
+
+import scala.{Char, Int, Unit}
+import scala.Predef.{assert, ArrowAssoc}
+
+import org.junit.Test
+
+class WithFilterTest {
+
+  @Test
+  def iterables(): Unit = {
+    val xs1 =
+      for {
+        x <- List(1, 2, 3)
+        if x % 2 == 0
+      } yield x + 1
+    val xs1t: List[Int] = xs1
+    assert(xs1 == List(3))
+  }
+
+  @Test
+  def maps(): Unit = {
+    val xs1 =
+      for {
+        (k, v) <- Map(1 -> 'a', 2 -> 'b', 3 -> 'c')
+        if k % 2 == 0
+      } yield (v, k)
+    val xs1t: Map[Char, Int] = xs1
+    assert(xs1 == Map('b' -> 2))
+  }
+
+  @Test
+  def sorted(): Unit = {
+    val xs1 =
+      for {
+        x <- TreeSet(1, 2, 3)
+        if x % 2 == 0
+      } yield x + 1
+    val xs1t: TreeSet[Int] = xs1
+    assert(xs1 == TreeSet(3))
```

Comment:
This test fails on Dotty. This is probably something related to hashcodes. We observed a weird behavior [here](https://github.com/scala/collection-strawman/pull/48#issuecomment-311615632) too. I’ll create an issue in Dotty.